docs/upgrading-to-v0.18.x.md

```diff
@@ -0,0 +1,244 @@
+filling required fields using the information from `my-nodepools.yaml` and
+`my-gkeclusters.yaml` files that we created in the beginning of migration steps.
+
+**Template to import `Cluster`:**
+
+```
+apiVersion: container.gcp.crossplane.io/v1beta2
+kind: Cluster
+metadata:
+  name: <to-be-filled>
+  annotations:
+    crossplane.io/external-name: <to-be-filled>
+spec:
+  # Consider setting deletionPolicy as "Orphan" to make sure you don't delete
+  # the cloud resource until you successfully validate the import
+  # deletionPolicy: Orphan
+  forProvider:
+    location: <to-be-filled>
+  providerConfigRef:
+    name: <to-be-filled>
+```
+
+**Template to import `NodePool`:**
+
+```
+apiVersion: container.gcp.crossplane.io/v1beta1
+kind: NodePool
+metadata:
+  name: <to-be-filled>
+  annotations:
+    crossplane.io/external-name: <to-be-filled>
+spec:
+  # Consider setting deletionPolicy as "Orphan" to make sure you don't delete
+  # the cloud resource until you successfully validate the import
+  # deletionPolicy: Orphan
+  forProvider:
+    cluster: <to-be-filled>
+  providerConfigRef:
+    name: <to-be-filled>
+```
+
+##### Steps
+
+1. Create resources using the above manifests
+2. Verify that resources are `READY` and `SYNCED`
+
+#### Importing as Beta
+
+**Template to import `Cluster`:**
+
+```
+apiVersion: container.beta.gcp.crossplane.io/v1beta1
+kind: Cluster
+metadata:
+  name: <to-be-filled>
+  annotations:
+    crossplane.io/external-name: <to-be-filled>
+spec:
+  # Consider setting deletionPolicy as "Orphan" to make sure you don't delete
+  # the cloud resource until you successfully validate the import
+  # deletionPolicy: Orphan
+  forProvider:
+    location: <to-be-filled>
+  providerConfigRef:
+    name: <to-be-filled>
+```
+
+**Template to import `NodePool`:**
+
+```
+apiVersion: container.beta.gcp.crossplane.io/v1alpha1
+kind: NodePool
+metadata:
+  name: <to-be-filled>
+  annotations:
+    crossplane.io/external-name: <to-be-filled>
+spec:
+  # Consider setting deletionPolicy as "Orphan" to make sure you don't delete
+  # the cloud resource until you successfully validate the import
+  # deletionPolicy: Orphan
+  forProvider:
+    cluster: <to-be-filled>
+  providerConfigRef:
+    name: <to-be-filled>
+```
+
+##### Steps
+
+1. Deploy [provider-gcp-beta] on the cluster.
+2. Create `ProviderConfig` resource for [provider-gcp-beta].
+3. Create resources using the above manifests.
+4. Verify that resources are `READY` and `SYNCED`.
+
+### Updating compositions
+
+If you have compositions relying on the deprecated types, you would need to
+update them as well to use the new types with proper fields and configuration
+depending on whether you have chosen `beta` or `stable`. Until you switch
+using the new types, your composite resources would fail since the old types no
+longer exist.
+
+Once you switch to the new types in your composition, your composite resources
```

Comment:
I think they need to do some surgery here and scaling down Crossplane deployment to 0 would be safer against recreation. Also a list of steps would be great, like
* Scale down Crossplane to 0
* Update `Composition`
* Mark existing managed resources as `deletionPolicy: Orphan`
* Back up managed resoruces
* Delete managed resoruces from cluster.
* Import new ones.
* Update `resourceRefs` with the new managed resources.
* Scale up Crossplane deployment to 1.

Reply:
Yeah, that's true. And actually, things get a bit more complicated if configuration packages are used to deploy compositions, e.g. how configuration package would be updated to new composition if crossplane is not running and when you scale it back to 1 which controller will process first, composition or packagerevision.

But, let me give it another thought if I can find something to add steps to the original flow considering compositions.

Reply:
@muvaf I think we already solved most of the problems that I was originally thinking and the only remaining one was preventing composing new resources. Added steps for this, please see the latest commit for the diff.